pyresample/geometry.py

```diff
@@ -717,37 +717,282 @@ def _do_transform(src, dst, lons, lats, alt):
+            lats.attrs = self.lats.attrs.copy()
+            lons.attrs = self.lons.attrs.copy()
+            try:
+                resolution = lons.attrs['resolution'] / ((x + y) / 2)
+                lons.attrs['resolution'] = resolution
+                lats.attrs['resolution'] = resolution
+            except KeyError:
+                pass
+
+        # Return the downsampled swath definition
+        return SwathDefinition(lons, lats)
+
+    def extend(self, x=0, y=0):
+        """Extend the swath definition along x (along-track) and y (across-track) dimensions.
+
+        By default, it does not extend on any direction.
+        To extend of n pixel  on both sides of the across-track direction, call swath_def.extend(x=0, y=2).
+        """
+        import xarray as xr
+
+        # Check input validity
+        x = int(x)
+        y = int(y)
+        if x < 0 or y < 0:
+            raise ValueError('x and y arguments must be positive integers.')
+
+        # Return SwathDefinition if nothing to extend
+        if x == 0 and y == 0:
+            return self
+
+        # Get lats/lons numpy arrays
+        src_lats, src_lats_format = _convert_2D_array(self.lats, to='numpy', dims=['y', 'x'])
+        src_lons, src_lons_format = _convert_2D_array(self.lons, to='numpy', dims=['y', 'x'])
+
+        dst_lats = src_lats
+        dst_lons = src_lons
+
+        # Extend on y direction (side0 and side2)
+        if y > 0:
+            list_side0 = (src_lons[1, :], src_lats[1, :], src_lons[0, :], src_lats[0, :])
+            list_side2 = (src_lons[-2, :], src_lats[-2, :], src_lons[-1, :], src_lats[-1, :])
+            extended_side0_lonlats = _get_extended_lonlats(*list_side0, npts=y)
+            extended_side2_lonlats = _get_extended_lonlats(*list_side2, npts=y)
+            dst_lats = np.concatenate((extended_side0_lonlats[1][::-1, :], dst_lats), axis=0)
+            dst_lats = np.concatenate((dst_lats, extended_side2_lonlats[1]), axis=0)
+            dst_lons = np.concatenate((extended_side0_lonlats[0][::-1, :], dst_lons), axis=0)
+            dst_lons = np.concatenate((dst_lons, extended_side2_lonlats[0]), axis=0)
+
+        # Extend on x direction (side1 and side3)
+        if x > 0:
+            list_side1 = (dst_lons[:, -2], dst_lats[:, -2], dst_lons[:, -1], dst_lats[:, -1])
+            list_side3 = (dst_lons[:, 1], dst_lats[:, 1], dst_lons[:, 0], dst_lats[:, 0])
+            extended_side1_lonlats = _get_extended_lonlats(*list_side1, npts=x, transpose=False)
+            extended_side3_lonlats = _get_extended_lonlats(*list_side3, npts=x, transpose=False)
+            dst_lats = np.concatenate((dst_lats, extended_side1_lonlats[1]), axis=1)
+            dst_lats = np.concatenate((extended_side3_lonlats[1][:, ::-1], dst_lats), axis=1)
+            dst_lons = np.concatenate((dst_lons, extended_side1_lonlats[0]), axis=1)
+            dst_lons = np.concatenate((extended_side3_lonlats[0][:, ::-1], dst_lons), axis=1)
```

Comment:
I think there is some extrapolation capabilities in python_geotiepoints, could we use these to avoid duplication?

Reply:
I couldn't find the code. Could you point me to it?

Reply:
https://github.com/pytroll/python-geotiepoints/blob/main/geotiepoints/interpolator.py#L110

Reply:
I noticed really strange behaviours when using fill_borders. Look at the code snippet in the PR